### ISSUE TITLE
Profile Zeebe to root cause cycle time degradation

### ISSUE THREAD

**Issue by korthout** [zeebe-io/zeebe#5314]
**Description**

The cycle time (performance) degraded somewhere between `v0.20.x` and `v0.21.x` (See: #5132). We should profile Zeebe to find the root cause.

**Comment by korthout** [comment]
There are multiple indicators that we are running into too many cpu cache misses (see: #4249 as main issue). To make sure that this problem still exists (AND is the reason for the cycle time increase (performance degradation) in my current benchmarks), we should run the same [async profiler](https://github.com/jvm-profiling-tools/async-profiler) again.

To get the cache misses profile, we can run: 
```
./profiler.sh -e cache-misses  -d 30 -i 1ms -o traces -f outputfile.txt jps``` 

This uses `jps` to find the active java process to profile. Both `jps` and `async-profiler` must somehow be installed on the zeebe containers to work.

**Comment by korthout** [comment]
The async profiler is not easily run against zeebe nodes deployed on k8s in gke. This is because on gke there is no direct access to the perf_events. I've tried to get access to them, but I've not been able so far. A workaround is to profile using itimer events, which allows to produce a flamegraph of the code Zeebe spends time on, but no cache-misses reporting. 

Therefore, we've profiled Zeebe `0.20.2` and `0.24.2` on gke using the itimer events AND profiled the cache-misses running Zeebe locally.

Flamegraph of Zeebe 0.20.2:
<img width="600" alt="Flamegraph of Zeebe 0.20.2" src="https://user-images.githubusercontent.com/3511026/94418283-788f5380-0181-11eb-9ac8-903f3044db59.png">

Flamegraph of Zeebe 0.24.2:
<img width="600" alt="Flamegraph of Zeebe 0.24.2" src="https://user-images.githubusercontent.com/3511026/94418368-96f54f00-0181-11eb-80f9-6ab6b7fc73f5.png">

We can see that in 0.24.2, Zeebe spends most of its time reading the log stream, while this was not the case in the 0.20.2. A major difference between the versions is that 0.20 still used our own code for reading the log stream (actually a separate log stream than the atomix log stream), while 0.24 only uses the atomix log stream. The old log stream was read with multiple optimisations, like memory mapped files and read ahead. Although read ahead is somewhat implemented in our usage of the atomix log stream, memory mapped files is disabled by default. 

The locally run cache-misses report further showed an increase of 3.5 times more cache misses between the versions. This might be completely caused by the increased usage of the atomix log stream, since reading always uses a FileChannel.

We should further investigate how we can fully support memory mapped files in the current log stream implementation.
